config/desc/imsim/instcat.py

```diff
@@ -0,0 +1,627 @@
+            full_name = os.path.join(self.sed_dir, name)
+            if not os.path.isfile(full_name):
+                full_name = os.path.join(self.inst_dir, name)
+            if not os.path.isfile(full_name):
+                raise OSError("Could not find file %s in either %s or %s"%(
+                              name, self.sed_dir, self.inst_dir))
+            sed = galsim.SED(full_name, wave_type='nm', flux_type='flambda')
+            sed = sed.withMagnitude(0, self._bp500)  # Normalize to mag 0
+            self._sed_cache[name] = sed
+
+        # TODO: Handle the dust effects.  internal, then redshift, then galactic.
+        #       Also probably need to think about the normalization and how that interacts with
+        #       the flux and/or mag.
+        return sed.atRedshift(redshift)
+
+    def getLens(self, index):
+        # The galsim.lens(...) function wants to be passed reduced
+        # shears and magnification, so convert the WL parameters as
+        # defined in phosim instance catalogs to these values.  See
+        # https://github.com/GalSim-developers/GalSim/blob/releases/1.4/doc/GalSim_Quick_Reference.pdf
+        # and Hoekstra, 2013, http://lanl.arxiv.org/abs/1312.5981
+        gamma1, gamma2, kappa = self.lens[index]
+        g1 = gamma1/(1. - kappa)   # real part of reduced shear
+        g2 = gamma2/(1. - kappa)   # imaginary part of reduced shear
+        mu = 1./((1. - kappa)**2 - (gamma1**2 + gamma2**2)) # magnification
+        return g1,g2,mu
+
+    def getDust(self, index):
+        params = dust[index]
+        if params[0].lower() != 'none':
+            internal_av = float(params[1])
+            internal_rv = float(params[2])
+            params = params[3:]
+        else:
+            internal_av = 0.
+            internal_rv = 0.
+            params = params[1:]
+
+        if params[0].lower() != 'none':
+            galactic_av = float(params[1])
+            galactic_rv = float(params[2])
+        else:
+            galactic_av = 0.
+            galactic_rv = 0.
+
+        return internal_av, internal_rv, galactic_av, galactic_rv
+
+    def getObj(self, index, gsparams=None, rng=None, bandpass=None, chromatic=False, exp_time=30):
+        params = self.objinfo[index]
+
+        magnorm = self.getMagNorm(index)
+        if magnorm >= 50:
+            # Mark of invalid object apparently
+            return None
+
+        if gsparams is not None:
+            gsparams = galsim.GSParams(**gsparams)
+
+        # Make the object according to the values in the objinfo
+
+        # Note: params here starts at 12, so all indices are 12 less than in Jim's code.
+        if params[0].lower() == 'point':
+            obj = galsim.DeltaFunction(gsparams=gsparams)
+
+        elif params[0].lower() == 'sersic2d':
+            a = float(params[1])
+            b = float(params[2])
+            if b > a:
+                # Invalid, but Jim just let's it pass.
```

Comment:
I think most of the comments in this file attributed to my work was actually done by Scott.  In any case, I suggest changing to something like `# Invalid, but the original imSim version just let it pass`

Reply:
Done.  Reworded all of these to refer to "previous code".